app/models/user.rb

```diff
@@ -4,19 +4,37 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
-
+  has_many :friendships, ->(user) { unscope(:where).where(user: user).or(where(friend: user)) }
+ 
   validates :first_name, length: { minimum: 3 }
   validates :first_name, :email, :username, presence: true
   validates :email, uniqueness: true
   validates :password, confirmation: { case_sensitive: true }
 
+  scope :confirmed, -> { where(confirmed: true) }
+  scope :pending, -> { where(confirmed: false) }
```

Comment:
Where are you using these scopes?

Reply:
Is it even valid? Do you have `confirmed` attribute in users table?

Reply:
There is nothing like these . These confirmed is in friendship table . I remove it from here.